### PR TITLE
boilerplate codes of (XGBoost)Controller

### PR DESCRIPTION
**Pull request by sperlingxx**
Add necessary boilerplate codes for a job controller, which aren't specific for XGBoost.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/xgboost/6)

### CODE REVIEW

**Comment by merlintang** [comment]
@terrytangyuan Thanks Terry, some codes would be removed and I suggest to add tavis CI as well.

File: pkg/controller.v1alpha1/xgboost/controller.go [hunk 1]
```diff
@@ -13,3 +13,294 @@
...
+		}
+		return true
+	}
+
+	xc.WorkQueue.AddRateLimited(key)
+
+	return true
+}
+
+func (xc *XGBoostController) enqueueXGBoostJob(job interface{}) {
```
Comment (merlintang) [inline pkg/controller.v1alpha1/xgboost/controller.go]: can you move this function to job.go

File: cmd/xgboost-operator.v1alpha1/app/server.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	}
+
+	// Get kubernetes config.
+	kcfg, err := clientcmd.BuildConfigFromFlags(opt.MasterURL, opt.Kubeconfig)
+	if err != nil {
+		log.Fatalf("Error building kubeconfig: %s", err.Error())
+	}
+
+	// Create clients.
+	kubeClientSet, leaderElectionClientSet, pytorchJobClientSet, kubeBatchClientSet, err := createClientSets(kcfg)
```
Comment (merlintang) [inline cmd/xgboost-operator.v1alpha1/app/server.go]: change pytorchjob name

File: pkg/controller.v1alpha1/xgboost/controller.go [hunk 3]
```diff
@@ -13,3 +13,294 @@
...
+// Run will set up the event handlers for types we are interested in, as well
+// as syncing informer caches and starting workers. It will block until stopCh
+// is closed, at which point it will shutdown the workqueue and wait for
+// workers to finish processing their current work items.
+func (xc *XGBoostController) Run(threadiness int, stopCh <-chan struct{}) error {
+	defer utilruntime.HandleCrash()
+	defer xc.WorkQueue.ShutDown()
+
+	// Start the informer factories to begin populating the informer caches.
+	log.Info("Starting TFJob controller")
```
Comment (merlintang) [inline pkg/controller.v1alpha1/xgboost/controller.go]: TFJob to XGBoostJob

**Comment by terrytangyuan** [comment]
Okay @sperlingxx please ping me when this is ready for review. @merlintang I've created https://github.com/kubeflow/xgboost-operator/issues/9 to track Travis CI as a separate item.

**Comment by sperlingxx** [comment]
@merlintang @terrytangyuan I think it's ready now.

**Pull request merged**